core/platform/lf_spike_support.c

```diff
@@ -0,0 +1,108 @@
+/*************
+Copyright (c) 2021, The University of California at Berkeley.
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+/** Baremetal RISC-V support for the C target of Lingua Franca.
+ *  
+ *  @author{Efsane Soyer <efsanesoyer@berkeley.edu>}
+ */
+
+#include "lf_spike_support.h"
+#include "../platform.h"
+#include "../tag.h" //needed for BILLION. In flexpret-script under tag.h CLOCK_FREQ is defined, but it's not defined in lf repo 
+
+#ifdef NUMBER_OF_WORKERS
+#if __STDC_VERSION__ < 201112L || defined (__STDC_NO_THREADS__) // (Not C++11 or later) or no threads support
+#include "lf_POSIX_threads_support.c"
+#else
+#include "lf_C11_threads_support.c"
+#endif
+#endif
+
+#include <stdio.h> // TODO: remove
+
+// FIXME: This assumption about clock frequency
+// varies by hardware platforms.
+//hardcoded because it wasn't defined in tag.h
+#define CLOCK_FREQ 100000000LL
```

Comment:
It seems like this should be a [target parameter](https://www.lf-lang.org/docs/handbook/target-declaration#summary-of-parameters).

Reply:
Can move this to the Makefile; Should I then have the code generator dynamically edit the Makefile?